### PR TITLE
[4.0] Remove JError::raise code

### PR DESCRIPTION
**Pull request by laoneo**
As discussed in #16952, this pr convertes the code which is calling `JError::raise*` in core. Next step would then be to remove the JError class.

Pinging @mbabker and @wilsonge for feedback.

### CODE REVIEW

File: administrator/components/com_banners/Helper/BannersHelper.php [hunk 1]
```diff
@@ -146,7 +146,7 @@ public static function updateReset()
 			}
 			catch (\RuntimeException $e)
 			{
-				\JError::raiseWarning(500, $db->getMessage());
+				\JFactory::getApplication()->enqueueMessage($db->getMessage(), 'error');
```
Comment (wilsonge) [inline administrator/components/com_banners/Helper/BannersHelper.php]: @mbabker My memory fails me, is this going to give sql queries or not? I don't *think* it does but wanna be sure

Comment (wilsonge) [reply administrator/components/com_banners/Helper/BannersHelper.php]: Also does `$db->getMessage()` actually exist? I think this is a bug and should be `$e->getMessage()`

Comment (laoneo) [reply administrator/components/com_banners/Helper/BannersHelper.php]: True.

Comment (mbabker) [reply administrator/components/com_banners/Helper/BannersHelper.php]: Query exceptions only give the database's error message.  The full query is a separate property on our exception class that's not part of the message.

File: libraries/src/CMS/Application/CMSApplication.php [hunk 2]
```diff
@@ -883,13 +883,13 @@ public function login($credentials, $options = array())
 					switch ($authorisation->status)
 					{
 						case Authentication::STATUS_EXPIRED:
-							return \JError::raiseWarning('102002', \JText::_('JLIB_LOGIN_EXPIRED'));
```
Comment (wilsonge) [inline libraries/src/CMS/Application/CMSApplication.php:886]: What do these numbers represent?

Comment (mbabker) [reply libraries/src/CMS/Application/CMSApplication.php:886]: An arbitrary code created by somebody long before our time.

Comment (brianteeman) [reply libraries/src/CMS/Application/CMSApplication.php:886]: iirc there was a plan to have a series of joomla error codes beginning 10...

**Pull request merged**